examples/protocol_specification_ex/tac.yaml

```diff
@@ -57,3 +57,15 @@ ct:ErrorCode: |
   }
   ErrorCodeEnum error_code = 1;
 ...
+---
+reply:
+  register: [tac_error]
+  unregister: [tac_error]
+  transaction: [transaction_confirmation,tac_error]
+  cancelled: []
+  game_data: []
+  transaction_confirmation: []
+  tac_error: []
+roles: {agent, controller}
+end_states: [open, closed]
+...
```

Comment:
So from reply structure, `cancelled`, `game_data`, `transaction_confirmation`, `tac_error`, all terminate a dialogue. The latter two are just one move dialogues initiated by the controller. Right?

In which case, I suggest the following: `game_cancelled`, `game_data_sent`, `transaction_confirmed`, and `tac_error` for end_states of the different dialogues which essentially convey the different states of the game. 

Note, this messaging structure isn't the best because it includes 4 different types of dialogues between agent, controller (some one message dialogues). Probably a better structure would be the following:

register: [tac_error,  game_data, unregister, cancelled]
unregister: [tac_error]
game_data: [transaction]
transaction: [transaction_confirmation, tac_error]
cancelled: []
transaction_confirmation: [transaction]
tac_error: []

Reply:
In light of your comments, I think it should actually be :
```
  register: [tac_error, game_data, cancelled]
  unregister: [tac_error]
  transaction: [transaction_confirmation,tac_error]
  cancelled: []
  game_data: [transaction]
  transaction_confirmation: [transaction]
  tac_error: []
```
`unregister` can be sent by the user at any time, they themselves would have had to send a register before but this cannot be expressed in our reply structure.

Reply:
If `unregister` is always after `register` we can make it so it replies to `register`. This prevents single message dialogues (just `unregister`).

Reply:
Its not a reply though; it is just another message initiated by the same side which sends register.

Reply:
So this points at scenarios where you have a dialogue but another type of message can be sent "in reply" to an own message (unregister "in reply" to register, both sent by the agent).

Reply:
That is precisely what the notion of reply is. It's not about the sender/receiver of messages, but it is a relation over messages defining how one message relates to another message.

Reply:
With our current setup, replying to own message is already possible. There are no restriction when replying that you should reply to someone else's message. So we do not have to do anything to enable this.

Reply:
If one wants to reply, regardless of the sender of the message being replied to, the process is the same.

Reply:
I agree with what you say.